R/MsBackendTimsTof-functions.R

```diff
@@ -163,23 +150,105 @@ MsBackendTimsTof <- function() {
             tmp <- rbindFill(tmp, data.frame(frame = 0L))
         f <- factor(paste(tmp$frame, tmp$scan), levels = unique(ids))
         if (anyDuplicated(ids)) {
-            if (length(sd) == 1)
-                res[I] <- unname(split(tmp[, sd, drop], f)[ids])
+            if (length(columns) == 1)
+                res[I] <- unname(split(tmp[, columns, drop], f)[ids])
             else
-                res[I] <- unname(split.data.frame(as.matrix(tmp[, sd, drop]),
-                                                  f)[ids])
+                res[I] <- unname(
+                    split.data.frame(as.matrix(tmp[, columns, drop]), f)[ids])
         } else {
-            if (length(sd) == 1) {
-                res[I] <- unname(split(tmp[, sd, drop], f))
+            if (length(columns) == 1) {
+                res[I] <- unname(split(tmp[, columns, drop], f))
             } else {
-                res[I] <- unname(split.data.frame(as.matrix(tmp[, sd, drop]),
-                                                  f))
+                res[I] <- unname(
+                    split.data.frame(as.matrix(tmp[, columns, drop]), f))
             }
         }
     }
     res
 }
 
+#' Parallel processing version of the function above
+#'
+#' @noRd
+.get_tims_columns_p <- function(x, columns, drop = TRUE, BPPARAM = bpparam()) {
```

Comment:
It seems to me that this parallel version is not used in the code. Isn't its performance better?

Reply:
I did not yet try that. I assume it would perform better because it can read data in parallel from multiple files - but I would like to test this on different setups (windows, linux, mac) first.